pydocstringformatter/_formatting/formatters_numpydoc.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+from collections import OrderedDict
+
+from pydocstringformatter._formatting.base import NumpydocSectionFormatter
+
+
+class NumpydocSectionOrderingFormatter(NumpydocSectionFormatter):
+    """Change section order to match numpydoc guidelines."""
+
+    name = "numpydoc-section-order"
+
+    numpydoc_section_order = (
+        # We skip summary, deprecation warning, and extended
+        # summary
+        "Parameters",
+        "Attributes",
+        "Methods",
+        "Returns",
+        "Yields",
+        "Receives",
+        "Other Parameters",
+        "Raises",
+        "Warns",
+        "Warnings",
+        "See Also",
+        "Notes",
+        "References",
+        "Examples",
+    )
+
+    def treat_sections(
+        self, sections: OrderedDict[str, list[str]]
+    ) -> OrderedDict[str, list[str]]:
+        """Sort the numpydoc sections into the numpydoc order."""
+        new_sections = OrderedDict([sections.popitem(last=False)])
+        new_sections.update(
+            OrderedDict(
+                [
+                    (sec_name, sections[sec_name])
+                    for sec_name in sorted(
+                        sections.keys(),
+                        key=self.numpydoc_section_order.index,
+                    )
+                ]
+            )
+        )
+        return new_sections
+
+
+class NumpydocNameColonTypeFormatter(NumpydocSectionFormatter):
+    """Ensure proper spacing around the colon separating names from types."""
+
+    name = "numpydoc-name-type-spacing"
+
+    numpydoc_sections_with_parameters = (
+        "Parameters",
+        "Attributes",
+        "Returns",
+        "Yields",
+        "Receives",
+        "Other Parameters",
+        "See Also",
+    )
+
+    def treat_sections(
+        self, sections: OrderedDict[str, list[str]]
+    ) -> OrderedDict[str, list[str]]:
+        """Ensure proper spacing around the colon separating names from types."""
+        new_sections = OrderedDict([])
+        for section_name, section_lines in sections.items():
```

Comment:
Is there a particular why we don't replace lines in-place here? You do so in `NumpydocSectionSpacingFormatter`.

Reply:
Trying to avoid side-effect, I think?  Since I'm not consistent, I can change this to just modify the lines in their original lists and pass that back.

Reply:
Let's do that!

Reply:
Now in-place